keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuator.kt

```diff
@@ -174,25 +176,37 @@ class ResourceActuator(
     }
   }
 
-  private suspend fun <T : Any> ResourceHandler<*, T>.resolve(resource: Resource<out ResourceSpec>): Pair<T, T?> =
+  private suspend fun <T : Any> ResourceHandler<*, T>.resolve(resource: Resource<out ResourceSpec>, coid: String): Pair<T, T?> =
     supervisorScope {
       val desired = async {
         try {
-          desired(resource)
+          log.debug("before desired: ${resource.id}: [$coid]")
+          val result = desired(resource)
+          log.debug("after desired: ${resource.id}: [$coid]")
+          result
```

Comment:
here and below, what is the meaning of the stand alone `result`?

Reply:
AIUI, blocks passed to an `async` have to evaluate to a value:

```
val desired = async {
```


Before my change, the block passed to async would evaluate to the return value of `desired(resource)`.

However, if I just added a log statement afterwards, the block would evaluate to the return value of the log statement. So, I needed to add a line afterwards that would evaluate to the return value of `desired` to preserve the behavior.

The `result` variable is just a temp for doing this.